File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        container = dict()
        index = 0
        for n in nums:
            diff = target - n
            if diff in container:
                return [container[diff], index]
            container[n] = index
            index += 1

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        const int len = numbers.size();

        int left = 0, right = len - 1;

        while (left < right) {
            int sum = numbers[left] + numbers[right];
            
            if (sum == target) {
                // note: return index w/ -1
                return { left + 1, right + 1};
            }

            if (sum < target) {
                left++;
                continue;
            }

            if (sum > target) {
                right--;
                continue;
            }
        }

        return {};
    }
};
